app.js

```diff
@@ -1,15 +1,54 @@
 import Fastify from 'fastify';
+import functions from './connections.js';
 
 const logOptions = {
   level: process.env.logLevel || 'info',
 };
-
 const app = Fastify({
   logger: logOptions,
 });
 
+const { connectToDB, connectToDBToFind, connectToDbToFinAll } = functions;
+
+const schema = {
+  body: {
+    type: 'object',
+    required: ['id', 'name', 'rank', 'limitations'],
+    properties: {
+      id: { type: 'integer' },
+      name: { type: 'string' },
+      rank: { type: 'integer' },
+      limitations: { type: 'array' },
+    },
+
+  },
+};
+
 app.get('/health', (req, res) => {
   res.status(200).send('ok');
 });
 
+app.post('/soldiers', { schema }, async (req, res) => {
+  const soldier = req.body;
+  await connectToDB(soldier).then(() => {
+    res.status(201).send('soldier created');
+  });
+});
+
+app.get('/soldiers/:id', async (req, res) => {
+  const findSoldier = req.params.id;
```

Comment:
this is not `findSoldier`

Reply:
למה לא? זה נותן לי את הid של החייל שאני מחפשת

Reply:
I don't have very good English, but i think "`findSoldier` means verb - lets find it. maybe did you meant `soldierToFind`.
Moreover, this is not actually "soldier". it is only the id...
so `id`, `soldierId`, or `soldierIdToFind` (too much for me) make more sense.

Reply:
fixed